### ISSUE TITLE
Exception at net.osmand.core.jni.OsmAndCoreJNI.swig_module_init 

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/OsmAnd#15715]
```
Version  OsmAnd+ 4.3.0
14.11.2022 8:49:34
Apk Version : 4.3.0 4300
Exception occured in thread Thread[main,5,main] : 
java.lang.UnsatisfiedLinkError: No implementation found for void net.osmand.core.jni.OsmAndCoreJNI.swig_module_init() (tried Java_net_osmand_core_jni_OsmAndCoreJNI_swig_1module_1init and Java_net_osmand_core_jni_OsmAndCoreJNI_swig_1module_1init__)
	at net.osmand.core.jni.OsmAndCoreJNI.swig_module_init(Native Method)
	at net.osmand.core.jni.OsmAndCoreJNI.<clinit>(OsmAndCoreJNI.java:4510)
	at net.osmand.core.jni.OsmAndCoreJNI.new_WeatherTileResourcesManager_IObtainValueAsyncCallback(Native Method)
	at net.osmand.core.jni.WeatherTileResourcesManager$IObtainValueAsyncCallback.<init>(WeatherTileResourcesManager.java:474)
	at net.osmand.plus.plugins.weather.WeatherWidget$1.<init>(WeatherWidget.java:35)
	at net.osmand.plus.plugins.weather.WeatherWidget.<init>(WeatherWidget.java:35)
	at net.osmand.plus.plugins.weather.TemperatureWidget.<init>(TemperatureWidget.java:14)
	at net.osmand.plus.plugins.weather.WeatherPlugin.createMapWidgetForParams(WeatherPlugin.java:215)
	at net.osmand.plus.plugins.weather.WeatherPlugin.createWidgets(WeatherPlugin.java:188)
	at net.osmand.plus.plugins.PluginsHelper.createMapWidgets(PluginsHelper.java:563)
	at net.osmand.plus.views.mapwidgets.WidgetsInitializer.createAllControls(WidgetsInitializer.java:68)
	at net.osmand.plus.views.mapwidgets.WidgetsInitializer.createAllControls(WidgetsInitializer.java:147)
	at net.osmand.plus.views.mapwidgets.MapWidgetRegistry.registerAllControls(MapWidgetRegistry.java:332)
	at net.osmand.plus.views.layers.MapInfoLayer.registerAllControls(MapInfoLayer.java:156)
	at net.osmand.plus.views.layers.MapInfoLayer.setMapActivity(MapInfoLayer.java:83)
	at net.osmand.plus.views.MapLayers.setMapActivity(MapLayers.java:236)
	at net.osmand.plus.activities.MapActivity.onCreate(MapActivity.java:288)
	at android.app.Activity.performCreate(Activity.java:7824)
	at android.app.Activity.performCreate(Activity.java:7813)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1306)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
Version  OsmAnd+ 4.3.0
14.11.2022 8:49:34
Apk Version : 4.3.0 4300
Exception occured in thread Thread[Initializing app,5,main] : 
java.lang.NoClassDefFoundError: <clinit> failed for class net.osmand.core.jni.OsmAndCoreJNI; see exception in other thread
	at net.osmand.core.jni.interface_ICoreResourcesProvider.<init>(interface_ICoreResourcesProvider.java:54)
	at net.osmand.core.android.CoreResourcesFromAndroidAssets.<init>(CoreResourcesFromAndroidAssets.java:38)
	at net.osmand.core.android.CoreResourcesFromAndroidAssets.loadFromCurrentApplication(CoreResourcesFromAndroidAssets.java:286)
	at net.osmand.plus.views.corenative.NativeCoreContext.init(NativeCoreContext.java:42)
	at net.osmand.plus.AppInitializer.initOpenGl(AppInitializer.java:752)
	at net.osmand.plus.AppInitializer.startApplicationBackground(AppInitializer.java:627)
	at net.osmand.plus.AppInitializer.lambda$startApplication$4$net-osmand-plus-AppInitializer(AppInitializer.java:873)
	at net.osmand.plus.AppInitializer$$ExternalSyntheticLambda1.run(Unknown Source:2)
	at java.lang.Thread.run(Thread.java:919)
 ```

This exception occured right at startup without any interaction at all.

**Comment by sonora** [comment]
@cepprice , @vshcherb Related to a wrong setting of _USE_OPENGL_RENDER_, see my findings in https://github.com/osmandapp/OsmAnd/issues/15627#issuecomment-1313597882, https://github.com/osmandapp/OsmAnd/issues/15627#issuecomment-1315156832, and https://github.com/osmandapp/OsmAnd/issues/15627#issuecomment-1315334230.

**Comment by scaidermern** [comment]
This happens every few app starts and then leads to OpenGL getting disabled (I guess you already know this).

**Comment by cepprice** [comment]
@scaidermern How old is your build?

**Comment by scaidermern** [comment]
> @scaidermern How old is your build?

See the first post. I'm using 4.3.0 from the playstore.

**Comment by ivanPyrohivskyi** [comment]
@scaidermern please check how it works in last night build, looks like we already fixed it in https://github.com/osmandapp/OsmAnd/commit/4ad0760ca0d2eaf549083bb8aeb136a2d1ce450c

**Comment by sonora** [comment]
Is there any tangible reason why we do not reset
`settings.USE_OPENGL_RENDER.set(false);`
on systems where the OpenGL init fails?

For what it's worth, **I can even with latest builds still 100% produce the crash** if `USE_OPENGL_RENDER` is wrongfully set as` true`, (be it that came from a build where we could still explicitly set it, or if it was set automatically by some update process), and whenever I then downgrade to a non-OpenGL build of last week. Other than uninstalling OsmAnd, I can 100% remediate the situation by interim installing an OpenGL capable build and simply using it to reset to "Version 1".

You may not consider that a valid use case worth fixing, but why take the risk? Why not be on the safe side and 
- reset the setting if we discover it does not work in a particular setup? (like contained in @creppice's original fix https://github.com/osmandapp/OsmAnd/commit/06b66709b6cf63f1a3e2b753a3b555c2aa095e53)
- and why do we instead hide the renderer setting entirely in the UI so in support cases we have no idea what it's set at?

There may be more and undetected issues caused by a false setting, like we had so many already, which could all be fixed reliably and pro-actively by these two measures?

**Comment by vshcherb** [comment]
```
Is there any tangible reason why we do not reset
settings.USE_OPENGL_RENDER.set(false);
on systems where the OpenGL init fails?
```

What if it fails only once and after restart works ok

**Comment by sonora** [comment]
Like in baseball: 3 strikes is out ... as follows: We have `OPENGL_RENDER_FAILED`, could make that an integer:
- Reset to 0 if init successful, if it fails increment 0 -> 1 -> 2 -> 3.
- Retry init OpenGL upon each app start, regardless of USE_OPENGL_RENDER, until it reaches 3.

**Comment by vshcherb** [comment]
As we have more and more features related to opengl obviously each needs a check that opengl is available, Grey out should be visible if build has opengl support, if build doesn't then it could be clear indication that build is limited

**Comment by sonora** [comment]
Exactly! Isn' this the convention we should follow:

A grayed-out line indicates a menu option we want you to see, or you may anyway expect to see, but the option is not available under the current circumstances. This could be due to 
- other conflicting settings being set, 
- the device/system/setup not supporting it, 
- or the current build not supporting it.

If we hide it entirely in some cases, we inhibit any awareness about OpenGL which we deem increasingly important in the future?

**Comment by vshcherb** [comment]
Sounds true

**Comment by scaidermern** [comment]
> @scaidermern please check how it works in last night build, looks like we already fixed it in [4ad0760](https://github.com/osmandapp/OsmAnd/commit/4ad0760ca0d2eaf549083bb8aeb136a2d1ce450c)

Where can I find a nightly build with OpenGL support? The one mentioned at https://www.osmand.net/docs/versions/nightly_versions/#opengl-build has no OpenGL support.

**Comment by vshcherb** [comment]
Yes it is
